client/extensions/woocommerce/app/dashboard/required-pages-setup-view.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
+import ProgressBar from 'components/progress-bar';
+import SetupHeader from './setup-header';
+import { setUpStorePages } from 'woocommerce/state/sites/setup-choices/actions';
+
+class RequiredPagesSetupView extends Component {
+	static propTypes = {
+		setUpStorePages: PropTypes.func.isRequired,
+		site: PropTypes.shape( {
+			ID: PropTypes.number.isRequired,
+		} ),
+	};
+
+	componentDidMount = () => {
+		const { site } = this.props;
+		if ( site ) {
+			this.props.setUpStorePages( site.ID );
+		}
+	}
+
+	render = () => {
+		const { translate } = this.props;
+		return (
+			<div className="card dashboard__setup-wrapper">
+				<SetupHeader
+					imageSource={ '/calypso/images/extensions/woocommerce/woocommerce-store-creation.svg' }
+					imageWidth={ 160 }
+					title={ translate( 'Setting up your store' ) }
+					subtitle={ translate( 'Give us a minute and we\'ll move right along.' ) }
+				>
+					<ProgressBar value={ 100 } isPulsing />
```

Comment:
Yeah I think so. I wanted something to show that something is happening (which `isPulsing` is useful for), since it's a single request.